TestFoundation/TestNSScanner.swift

```diff
@@ -41,4 +43,19 @@ class TestNSScanner : XCTestCase {
         XCTAssert(scanner.scanFloat(&value), "A Float should be found in the string `-350000000000000000000000000000000000000000`.")
         XCTAssert(value.isInfinite, "Scanned Float value of the string `-350000000000000000000000000000000000000000` should be infinite`.")
     }
+
+    func test_scanString() {
+        let scanner = Scanner(string: "apple sauce")
+
+        guard let firstPart = scanner.scanString(string: "apple ") else {
+            XCTFail()
+            return
+        }
+
+        XCTAssertEqual(firstPart, "apple ")
+        XCTAssertFalse(scanner.atEnd)
+
+        let _ = scanner.scanString(string: "sauce")
+        XCTAssertTrue(scanner.atEnd)
```

Comment:
`isAtEnd` is supposed to return `true` if the scanner hasn't actually consumed all the input, but the remaining input can all be skipped (due to `charactersToBeSkipped`). It would be great to actually test that behavior.